src/Engine/Renderer/RenderTechnique/RenderTechnique.hpp

```diff
@@ -13,79 +13,174 @@ namespace Ra {
+     * @param pass The index of the pass
+     * @return The pass shader configuration
+     */
+    const ShaderConfiguration&
+    getConfiguration( Core::Utils::Index pass = DefaultRenderingPasses::LIGHTING_OPAQUE ) const;
+
+    /**
+     * Get the ShaderProgram associated with the pass
+     * @param pass The index of the pass
+     * @return  The pass shader program if the pass is configured, nullptr otherwise.
+     */
+    const ShaderProgram*
+    getShader( Core::Utils::Index pass = DefaultRenderingPasses::LIGHTING_OPAQUE ) const;
+
+    /**
+     * Set the shader parameter provider for the given pass.
+     * @note all passes must be configured first using RenderTechnique::setConfiguration() before
+     * setting the parameter provider for each pass.
+     * @param provider The rendering parameter provider
+     * @param pass The index of the pass to set. If this -1, all active passes will share the same
+     * provider.
+     */
+    void setParametersProvider( const std::shared_ptr<ShaderParameterProvider>& provider,
+                                Core::Utils::Index pass = Core::Utils::Index( -1 ) );
+
+    /**
+     * Get the ShaderProgram parameters associated with the pass
+     * @param pass The index of the pass
+     * @return  The pass ShaderParameterProvider if the pass is configured, nullptr otherwise.
+     */
+    const ShaderParameterProvider* getParametersProvider(
+        Core::Utils::Index pass = DefaultRenderingPasses::LIGHTING_OPAQUE ) const;
+
+    /**
+     * Update all openGL status of active passes.
+     */
     void updateGL();
-    bool shaderIsDirty( PassName pass = LIGHTING_OPAQUE ) const;
 
-    // creates a Radium default rendertechnique :
-    //      Z_PREPASS = DepthDepthAmbientPass
-    //      LIGHTING_OPAQUE = BlinnPhong
-    //      LIGHTING_TRANSPARENT = LitOIT
+    /**
+     * Test if the given pass is dirty (openGL state not updated)
+     * @param pass The index of the pass
+     * @return Tru if pass must be updated.
+     */
+    bool shaderIsDirty( Core::Utils::Index pass = DefaultRenderingPasses::LIGHTING_OPAQUE ) const;
+
+    /**
+     * Creates a default technique based on the ForwarRenderer sementic.
+     *  pass 1 --> Z_PREPASS
+     *  pass 0 --> LIGHTING_OPAQUE
+     *  pass 2 --> LIGHTING_TRANSPARENT
+     * @return the default technique
+     */
     static Ra::Engine::RenderTechnique createDefaultRenderTechnique();
 
   private:
-    using ConfigurationSet = std::map<PassName, ShaderConfiguration>;
-    using ShaderSet        = std::map<PassName, const ShaderProgram*>;
-    ConfigurationSet shaderConfig;
-    ShaderSet shaders;
-
-    std::shared_ptr<Material> material{nullptr};
-
-    // Change this if there is more than 8 configurations
-    unsigned char dirtyBits{Z_PREPASS | LIGHTING_OPAQUE | LIGHTING_TRANSPARENT};
-    unsigned char setPasses{NO_PASS};
+    /// Maximum number of passses in the technique
+    static constexpr int m_maxNbPasses{32};
```

Comment:
I was asked to use `s_` prefix on another PR by @nmellado and I do agree with this, it differentiates the object attributes from the class static members. Maybe we should update the conventions regarding this.